src/psyclone/tests/domain/gocean/transformations/gocean_extract_test.py

```diff
@@ -672,11 +677,47 @@ def test_rename_region(tmpdir):
                  {'create_driver': True, 'region_name': ("main", "update")})
 
     # Test that the extraction code contains the right names
-    assert 'CALL psy_data%PreStart("main", "update", 4, 3)' in str(psy.gen)
+    assert 'CALL extract_psy_data%PreStart("main", "update", 4, 3)' \
+        in str(psy.gen)
 
     # Now test if the created driver has the right name, and will open the
     # right file:
     driver_name = tmpdir.join("driver-main-update.f90")
     with open(str(driver_name), "r") as driver_file:
         driver_code = driver_file.read()
-    assert 'CALL psy_data%OpenRead("main", "update")' in driver_code
+    assert 'CALL extract_psy_data%OpenRead("main", "update")' in driver_code
+
+
+# -----------------------------------------------------------------------------
+def test_change_prefix(tmpdir, monkeypatch):
+    '''
+    This tests that the prefix of an gocean extract transformation
+    can be changed, and that the new prefix is also used in the
+    created driver.
+    '''
+    # Use tmpdir so that the driver is created in tmp
+    tmpdir.chdir()
+
+    psy, invoke = get_invoke("single_invoke_scalar_float_arg.f90",
+                             GOCEAN_API, idx=0, dist_mem=False)
+
+    # In order to use a different prefix, this prefix needs to be valid.
+    # So monkeypatch the valid prefix names in the config object:
+    from psyclone.configuration import Config
+    config = Config.get()
+    monkeypatch.setattr(config, "_valid_psy_data_prefix", ["NEW"])
+
+    etrans = GOceanExtractTrans()
+    etrans.apply(invoke.schedule.children[0],
+                 {'create_driver': True, 'region_name': ("main", "update"),
+                  'class': "NEW"})
```

Comment:
Coming back to this 'cold', I found myself wondering what 'class' meant. Having looked at the documentation, I'm wondering whether we shouldn't call this 'prefix' or 'fortran_prefix'?

Reply:
I've renamed it to 'prefix' (avoiding the word Fortran in case that we might one way or another have C/... output in the future - the validity of the names should work in other languages as well). I also updated the docs somewhat to use 'class prefix' in some places instead of 'class'.